### PR TITLE
PLAT-140745: Refine logic to handle the last input type

### PR DESCRIPTION
**Pull request by 0x64**
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
The current API to handle the last input type imported by callback prop for `SpotlightRootDecorator` and it will be better to be imported in a normal way.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated to import key APIs from the module, not through callback function prop.
Also, updated TabLayout to use the API without querying a specific node to get the type info.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-140745

### CODE REVIEW

File: TabLayout/TabLayout.js [hunk 1]
```diff
@@ -38,8 +40,7 @@ const TabLayoutContext = createContext(null);
 const TouchableCell = Touchable(Cell);
 
 const isTouchMode = () => {
-	const rootContainer = document.querySelector('#root > div');
-	return rootContainer && rootContainer.classList.contains('spotlight-input-touch');
+	return getLastInputType() === 'touch';
```
Comment (jeonghee27) [inline TabLayout/TabLayout.js:43]: Please remove "{ return "

File: ThemeDecorator/ThemeDecorator.js [hunk 2]
```diff
@@ -296,4 +284,4 @@ const ThemeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 });
 
 export default ThemeDecorator;
-export {ThemeDecorator};
+export {ThemeDecorator, getInputType as getLastInputType};
```
Comment (jeonghee27) [inline ThemeDecorator/ThemeDecorator.js:287]: I'm not sure we should change the name

Comment (jeonghee27) [reply ThemeDecorator/ThemeDecorator.js:287]: Please rename when import

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: ThemeDecorator/ThemeDecorator.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ import {Component} from 'react';
 import classNames from 'classnames';
 import {ResolutionDecorator} from '@enact/ui/resolution';
 import {FloatingLayerDecorator} from '@enact/ui/FloatingLayer';
-import SpotlightRootDecorator from '@enact/spotlight/SpotlightRootDecorator';
+import SpotlightRootDecorator, {activateInputType, getInputType, setInputType} from '@enact/spotlight/SpotlightRootDecorator';
```
Comment (MikyungKim) [inline ThemeDecorator/ThemeDecorator.js:17]: I expected import `getInputType` as `getLastInputType` here.

File: ThemeDecorator/ThemeDecorator.js [hunk 4]
```diff
@@ -177,7 +177,7 @@ const ThemeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 		[css.bg]: !overlay
 	});
 
-	let spotlightInputType = {};
+	const spotlightInputType = {request: null};
```
Comment (MikyungKim) [inline ThemeDecorator/ThemeDecorator.js:180]: How about renaming this as below?
```suggestion
	const requestInputType = null;
```

**Review by MikyungKim — APPROVED** [review]
:lgtm:

**Pull request merged**